src/app/with-routing/my-page.component.ts

```diff
@@ -0,0 +1,39 @@
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+@Component({
+  selector: 'app-my-page',
+  template: `
+    <div class="title">{{ title }}</div>
+    <div class="dynamic-title">{{ dynamicTitle }}</div>
+    <div class="foo">{{ foo }}</div>
+    <div class="bar">{{ bar }}</div>
+    <div class="baz">{{ baz$ | async }}</div>
+    <a class="link-1" [routerLink]="['foo']">Some link</a>
+    <a class="link-2" (click)="navigate()">Other link</a>
+  `
+})
+export class MyPageComponent implements OnInit {
+  public title: string;
+  public dynamicTitle: string;
+  private foo?: string;
+  private bar?: string;
+  private baz$: Observable<string>;
+
+  constructor(private route: ActivatedRoute, private router: Router) {}
+
+  ngOnInit(): void {
```

Comment:
Note that you need to unsubscribe from each one on destroyed.

Reply:
👍